### PR TITLE
Fix issue of last exercise of the day `sleep()` passed end time

### LINKED ISSUE #20: Do not pull any new exercises after 1800 Eastern
So there is a `sleep()` call that runs at the last *valid* call at 1700, but it just *sleeps* and *then* runs the `assignExercise()` call. It just so happens that sometimes this `sleep()` runs over into 1800 which gets people angry... :(

I'll refactor when I have time.

https://github.com/TeachersPayTeachers/slackbot-workout/blob/master/slackbotExercise.py#L161

### PR DESCRIPTION
**Pull request by drincruz**
Obviously a **WIP** PR, mainly because it hasn't been *fully* tested. This should fix #20, but I also fixed some of the PEP8 as I made some changes, so this *looks* like a lot of changes, but it's not really. I'll update this PR when I'm content with my changes.

The issue of having the last exercise of the day run *after* the specified end time is fixed by checking if the current time + the next random interval is still within the hours that are specified.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -1,6 +1,11 @@
 {
   "debug": false,
 
+  "workoutTime": {
```
Comment (ScotterC) [inline config.json]: nice. love moving this to the config

File: slackbotExercise.py [hunk 2]
```diff
@@ -261,30 +262,61 @@ def saveUsers(bot):
         pickle.dump(bot.user_cache,f)
 
 
-def workout_time():
+def workout_time(bot):
     # Between 0900 and 1800; 9am - 6pm
```
Comment (ScotterC) [inline slackbotExercise.py]: can remove comment now

File: slackbotExercise.py [hunk 3]
```diff
@@ -261,30 +262,61 @@ def saveUsers(bot):
...
+            # If the next interval is within valid hours
+            if is_valid_interval(bot, sleep_interval):
+                # Get an exercise to do
+                exercise = select_exercise(bot)
+
+                # Announce the next lottery
+                announce_next_lottery_time(bot, exercise, sleep_interval)
+
+                # Sleep
+                time.sleep(sleep_interval)
```
Comment (ScotterC) [inline slackbotExercise.py]: makes sense to have all major time distortions (sleep etc) in the main loop and not hidden in sub methods

File: slackbotExercise.py [hunk 4]
```diff
@@ -261,30 +262,61 @@ def saveUsers(bot):
...
+
+                # Assign the exercise to someone
+                assignExercise(bot, exercise)
 
-            # Assign the exercise to someone
-            assignExercise(bot, exercise)
         if save_user_time():
```
Comment (ScotterC) [inline slackbotExercise.py]: Related to our time failures.  Looks like `save_user_time` hasn't worked at all so far. Any idea why the time isn't every equating to the save_user_time ?

File: slackbotExercise.py (hunk 2 above)
Comment (drincruz) [inline slackbotExercise.py]: @ScotterC comment removed. good call! :)

File: slackbotExercise.py (hunk 4 above)
Comment (drincruz) [inline slackbotExercise.py]: @ScotterC not quite sure, i'll take a look into this more later.

**Comment by drincruz** [comment]
@ScotterC i've made updates and i've also tested (locally). LGTM. :+1: ;)

**Pull request merged**